Number-of-Island.java

```diff
@@ -0,0 +1,97 @@
+## Problem1 (https://leetcode.com/problems/number-of-islands/)
+Given an m x n 2D binary grid grid which represents a map of '1's (land) and '0's (water), return the number of islands.
+
+An island is surrounded by water and is formed by connecting adjacent lands horizontally or vertically. You may assume all four edges of the grid are all surrounded by water.
+
+// Approach : This question can be done with both BFS and DFS.
+// BFS approach:
+// In this we want to give no of islands so we will run a loop and find a 1 and then we will add it in a queue and we will do a traversal and search if there is 1 at up,down,left, right if so we will add it in a queue.
+// We will carry out BFS traversal for all islands.
+// Also we don't need to keep track of levels so we don't need a size variable. For every island we will do a seperate dfs so we can keep a variable and return that
+// Time Complexity : O(m*n) -> as we are visiting every node
+// Space Complexity : O(m+n) -> the queue will hold maximum m+n elements in worst case 
```

Comment:
In worst case, the queue will hold m*n elements. So the Space complexity worst case is O(m*n).